### PR TITLE
Refactored Concourse creation Terraform

### PR DESCRIPTION
**Pull request by vijay-veeranki-moj**
Refactored Concourse creation Terraform using helm provider and updated README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,7 +7,15 @@ In order to setup concourse initially on a cluster, you will need to have `terra
 
 1. Edit `resources/secrets.tf` and add a configuration block for the new cluster, if one does not already exist.
 
-2. `./bin/setup-on-cluster <cluster-name>` where `cluster-name` is the name of the cluster you want to install Concourse CI on, as it appears in the terraform workspaces [here](https://github.com/ministryofjustice/kubernetes-investigations/tree/master/terraform/cloud-platform).
+## Prerequisite before running step 2.
+a) Select the name of the cluster you want to install Concourse CI on, as it appears in the terraform workspaces [here](https://github.com/ministryofjustice/kubernetes-investigations/tree/master/terraform/cloud-platform).
+
+kubectl config use-context <cluster-name>
```
Comment (alkar) [inline README.md]: Typically, this would be `<cluster-name>.k8s.integration.dsd.io`

File: README.md [hunk 2]
```diff
@@ -7,7 +7,15 @@ In order to setup concourse initially on a cluster, you will need to have `terra
 
 1. Edit `resources/secrets.tf` and add a configuration block for the new cluster, if one does not already exist.
 
-2. `./bin/setup-on-cluster <cluster-name>` where `cluster-name` is the name of the cluster you want to install Concourse CI on, as it appears in the terraform workspaces [here](https://github.com/ministryofjustice/kubernetes-investigations/tree/master/terraform/cloud-platform).
+## Prerequisite before running step 2.
+a) Select the name of the cluster you want to install Concourse CI on, as it appears in the terraform workspaces [here](https://github.com/ministryofjustice/kubernetes-investigations/tree/master/terraform/cloud-platform).
```
Comment (alkar) [inline README.md]: Update the link to point to `cloud-platform-infrastructure` - `kubernetes-investigations` is the old name.

File: bin/setup-on-cluster [hunk 3]
```diff
@@ -1,39 +1,6 @@
 #!/bin/sh
```
Comment (alkar) [inline bin/setup-on-cluster]: Can we please get rid of this script completely?

File: resources/main.tf [hunk 4]
```diff
@@ -327,3 +322,23 @@ resource "kubernetes_secret" "concourse_basic_auth_credentials" {
     password = "${random_string.basic_auth_password.result}"
   }
 }
+
+resource "helm_release" "concourse_proxy" {
```
Comment (alkar) [inline resources/main.tf]: Rename to simply `concourse` (same with `name` attribute below)

File: README.md (hunk 2 above)
Comment (vijay-veeranki-moj) [reply README.md]: updated readme as suggested

File: README.md (hunk 1 above)
Comment (vijay-veeranki-moj) [reply README.md]: updated use-context to <cluster-name>.k8s.integration.dsd.io

File: bin/setup-on-cluster (hunk 3 above)
Comment (vijay-veeranki-moj) [reply bin/setup-on-cluster]: removed bin/setup-on-cluster

File: resources/main.tf (hunk 4 above)
Comment (vijay-veeranki-moj) [reply resources/main.tf]: added it by mistake, now removed renamed to concourse

File: README.md [hunk 5]
```diff
@@ -7,7 +7,16 @@ In order to setup concourse initially on a cluster, you will need to have `terra
...
+## Prerequisite before running step 2.
+   Select the name of the cluster you want to install Concourse CI on, as it appears in the terraform workspaces [here](https://github.com/ministryofjustice/cloud-platform-infrastructure/tree/master/terraform/cloud-platform).
+
+   kubectl config use-context <cluster-name>.k8s.integration.dsd.io
+
+2. Run below commands to apply the changes required to reach the desired state of the configuration.
+   cd resources
+   terraform workspace select <cluster-name> 
+   terraform init 
+   terraform apply -auto-approve | sed -E 's/((content|template):[[:space:]]+)".+"/\1<REDACTED>/'
```
Comment (alkar) [inline README.md]: I guess we don't really need this `sed` anymore?

File: README.md [hunk 6]
```diff
@@ -7,7 +7,16 @@ In order to setup concourse initially on a cluster, you will need to have `terra
 
 1. Edit `resources/secrets.tf` and add a configuration block for the new cluster, if one does not already exist.
 
-2. `./bin/setup-on-cluster <cluster-name>` where `cluster-name` is the name of the cluster you want to install Concourse CI on, as it appears in the terraform workspaces [here](https://github.com/ministryofjustice/kubernetes-investigations/tree/master/terraform/cloud-platform).
+## Prerequisite before running step 2.
```
Comment (alkar) [inline README.md]: I'm re-reading this again. I think this entire section could do with some re-writing (feels more complicated than it should be). Essentially:
1. `kubectl config use-context`
2. `terraform workspace select`
3. edit `secrets.tf` if needed
4. run terraform

(as a side note, I think we should stop repeating the `terraform init / plan/ apply` song in every other repo :) )

Comment (vijay-veeranki-moj) [reply README.md]: Agree, will update the README, Thanks

Comment (vijay-veeranki-moj) [reply README.md]: > I'm re-reading this again. I think this entire section could do with some re-writing (feels more complicated than it should be). Essentially:
> 
> 1. `kubectl config use-context`
> 2. `terraform workspace select`
> 3. edit `secrets.tf` if needed
> 4. run terraform
> 
> (as a side note, I think we should stop repeating the `terraform init / plan/ apply` song in every other repo :) )

Update the README as suggested

**Pull request merged**